battle.cpp

```diff
@@ -592,18 +609,30 @@ void Game::GetSides(ARegion *r, AList &afacs, AList &dfacs, AList &atts,
 	int noaida = 0, noaidd = 0;
 	for (int i=-1;i<j;i++) {
 		ARegion * r2 = r;
-		if (i>=0) {
+		bool fromNeighboar = i >= 0;
+
+		if (fromNeighboar) {
+			// get neighbor region
 			r2 = r->neighbors[i];
+
+			// look for another neighbor if in current direction not neighbor exist
 			if (!r2) continue;
-			forlist(&r2->objects) {
-				/* Can't get building bonus in another region */
-				((Object *) elem)->capacity = 0;
-				((Object *) elem)->shipno = ((Object *) elem)->ships.Num();
-			}
-		} else {
-			forlist(&r2->objects) {
-				Object * o = (Object *) elem;
-				/* Set building capacity */
+		}
+
+		forlist (&r2->objects) {
+			Object * o = (Object *) elem;
+			bool extendFortDefence =
```

Comment:
this logic seems can be improved. Can you create if else statement with such condition

```
if (Globals->EXTENDED_FORT_DEFENCE_COVERAGE && o->IsBuilding() && !o->IsFleet()) {
  ... add bonus ...
} else {
   ... do not add bonus ...
}
```

so you do not need to clear anything and bonus is added only if it passes all tests.

Reply:
I'll look at what I can do